lib/extensions/immutable_raise_error.ex

```diff
@@ -163,22 +163,71 @@ defmodule AshPostgres.Extensions.ImmutableRaiseError do
       {nil, row_token} ->
         {:ok,
          Ecto.Query.dynamic(
-           fragment("ash_raise_error_immutable(?::jsonb, ?)", ^encoded, ^row_token)
+           fragment("ash_raise_error_immutable(?::jsonb, ?)", ^error_payload, ^row_token)
          ), acc}
 
       {dynamic_type, row_token} ->
         {:ok,
          Ecto.Query.dynamic(
            fragment(
              "ash_raise_error_immutable(?::jsonb, ?, ?)",
-             ^encoded,
+             ^error_payload,
              ^dynamic_type,
              ^row_token
            )
          ), acc}
     end
   end
 
+  # Encodes an error payload as jsonb using only IMMUTABLE SQL functions.
+  #
+  # Strategy:
+  # * Split the 'input' into Ash expressions and literal values
+  # * Build the base json map with the exception name and literal input values
+  # * For each expression value, use nested calls to `jsonb_set` (IMMUTABLE) to add the value to
+  #   'input', converting each expression to jsonb using `ash_to_jsonb_immutable` (which pins
+  #   session GUCs for deterministic encoding)
+  defp expression_error_payload(exception, input, query, bindings, acc) do
+    {expr_inputs, literal_inputs} =
+      Enum.split_with(input, fn {_key, value} -> Ash.Expr.expr?(value) end)
+
+    base_json = %{exception: inspect(exception), input: Map.new(literal_inputs)}
+
+    Enum.reduce(expr_inputs, {base_json, acc}, fn
+      {key, expr_value}, {current_payload, acc} ->
+        path_expr = %Ash.Query.Function.Type{
+          arguments: [["input", to_string(key)], {:array, :string}, []]
+        }
+
+        new_value_jsonb =
+          %Ash.Query.Function.Fragment{
+            arguments: [raw: "ash_to_jsonb_immutable(", expr: expr_value, raw: ")"]
+          }
+
+        {%Ecto.Query.DynamicExpr{} = new_payload, acc} =
+          AshSql.Expr.dynamic_expr(
+            query,
+            %Ash.Query.Function.Fragment{
+              arguments: [
+                raw: "jsonb_set(",
```

Comment:
This is interesting 🤔 we can't use something like jsonb_build_object? We have to build a big nested "set" expression?